zilliqa/src/consensus.rs

```diff
@@ -3208,6 +3221,21 @@ impl Consensus {
     pub fn get_sync_data(&self) -> Result<Option<SyncingStruct>> {
         self.sync.get_sync_data()
     }
+
+    /// This function is intended for use only by admin_forceSetView API. It is dangerous and should not be touched outside of testing or test network recovery.
+    ///
+    /// Force set our view and override exponential timeout value with given value for this view only
+    /// View value must be larger than current
+    pub fn force_set_view(&mut self, view: u64, timeout_mins: u64) -> Result<()> {
+        if self.get_view()? >= view {
+            return Err(anyhow!(
+                "view cannot be forced into lower view than current"
+            ));
+        }
+        self.set_view(view)?;
+        self.force_set_view = Some((view, Duration::from_secs(timeout_mins * 60)));
```

Comment:
```suggestion
        self.force_set_view = Some((view, Duration::from_mins(timeout_mins)));
```
but I guess this will change if we make @shawn-zil 's proposed change anyway.

Reply:
`Duration::from_` anything larger than secs is nightly